flask/helpers.py

```diff
@@ -553,7 +538,9 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
         rv.cache_control.max_age = cache_timeout
         rv.expires = int(time() + cache_timeout)
 
-    if add_etags and filename is not None:
+    if add_etags and filename is not None and file is None:
```

Comment:
Even with #1988 amending this, this particular line is still there and the `if` can never be `True`, because `file` is never `None`.
I don't know if I should just delete `and file is None` or if it should be more subtle.